programs/tollbooth/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+#![allow(unexpected_cfgs)] // warning: unexpected `cfg` condition value: `anchor-debug`
+
+use anchor_lang::prelude::*;
+
+declare_id!("7czSB69NDz3QJSveAga7igHR86ZkESEou9q845xzPkgf");
+
+const SNAPSHOT_SEED: &[u8] = b"snapshot";
+
+#[program]
+pub mod tollbooth {
+    use super::*;
+
+    #[instruction(discriminator = [0])]
+    pub fn enter<'info>(ctx: Context<'_, '_, '_, 'info, Enter<'info>>) -> Result<()> {
+        ctx.accounts.sponsor_snapshot.lamports = ctx.accounts.sponsor.lamports();
+        Ok(())
+    }
+
+    #[instruction(discriminator = [128])]
+    pub fn exit<'info>(
+        ctx: Context<'_, '_, '_, 'info, Exit<'info>>,
+        max_allowed_spending: u32,
+    ) -> Result<()> {
+        require_gte!(
+            ctx.accounts
+                .sponsor
+                .lamports()
+                .saturating_add(u64::from(max_allowed_spending)),
+            ctx.accounts.sponsor_snapshot.lamports,
```

Comment:
Super minor nit but I don't know about you but reading this seemed backwards to me, I think it's more obvious to compare `max_allowed_spending` to the difference between the value at enter and the current value than it is to do it this way